src/main/java/com/adobe/abp/regola/rules/RangeRule.java

```diff
@@ -0,0 +1,140 @@
+        return new EvaluationResult() {
+            private Result result = Result.MAYBE;
+            private Number evaluatedFact;
+            private String message;
+            private Throwable cause;
+
+            @Override
+            public RuleResult snapshot() {
+                return ValuesRuleResult.builder().with(r -> {
+                    r.type = getType();
+                    r.key = getKey();
+                    r.description = getDescription();
+                    r.result = result;
+                    r.expectedValue = getExpectedValue();
+                    r.actualValue = evaluatedFact;
+                    r.message = message;
+                    r.cause = cause;
+                    r.ignored = isIgnore();
+                }).build();
+            }
+
+            @Override
+            public CompletableFuture<Result> status() {
+                return factsResolver.resolveFact(getKey())
+                        .thenCompose(fact -> FutureUtils.supplyAsync(() -> {
+                            if (fact instanceof Number) {
+                                Number value = (Number) fact;
+                                if (isValidRange(value)) {
+                                    result = Result.VALID;
+                                } else {
+                                    result = Result.INVALID;
+                                    message = "Value is outside the range " + getExpectedValue();
+                                }
+                                evaluatedFact = value;
+                            } else {
+                                result = Result.INVALID;
+                                message = "Fact is not a Number";
+                            }
+                            return result;
+                        }, executor))
+                        .whenComplete((result, throwable) -> Optional.ofNullable(getAction())
+                                .ifPresent(action -> action.onCompletion(result, throwable, snapshot())))
+                        .exceptionally(throwable -> {
+                            result = Result.FAILED;
+                            message = throwable.getMessage();
+                            cause = throwable;
+                            return result;
+                        });
+            }
+
+            private String getExpectedValue() {
+                StringBuilder expectedValue = new StringBuilder();
+                if (inclusiveMin) {
+                    expectedValue.append('[');
+                } else {
+                    expectedValue.append('(');
+                }
+                expectedValue.append(min).append(", ").append(max);
+                if (inclusiveMax) {
+                    expectedValue.append(']');
+                } else {
+                    expectedValue.append(')');
+                }
+                return expectedValue.toString();
+            }
```

Comment:
The expected value in the `ValuesResultResult` is of type `T` (a generic), so we should be able to describe the range as an object rather than as a string.